content/docs/internals/health-checks.mdx

```diff
@@ -0,0 +1,280 @@
+      failureThreshold: 5
+```
+
+### Startup
+
+The startup endpoint `/startupz` waits for all components to report ready - in practice the latest version of the configuration should be synced before starting.
+
+### Readiness
+
+### Liveness
+
+The `/healthz` endpoint
+
+### Debugging
+
+If your pods are frequently restarted due to failed checks, the endpoints provide human-readable status reports from Pomerium. For example:
+
+```json
+{
+  "authenticate.service": {
+    "status": "RUNNING"
+  },
+  "authorize.service": {
+    "status": "RUNNING"
+  },
+  "config.databroker.build": {
+    "status": "RUNNING"
+  },
+  "databroker.sync.initial": {
+    "status": "RUNNING"
+  },
+  "envoy.server": {
+    "status": "RUNNING"
+  },
+  "proxy.service": {
+    "status": "RUNNING"
+  },
+  "storage.backend": {
+    "status": "RUNNING",
+    "attributes": [
+      {
+        "Key": "backend",
+        "Value": "in-memory"
+      }
+    ]
+  },
+  "xds.cluster": {
+    "status": "RUNNING"
+  },
+  "xds.listener": {
+    "status": "RUNNING"
+  },
+  "xds.route-configuration": {
+    "status": "RUNNING"
+  }
+}
+```
+
+which can help identify which component(s) are unhealthy and surface errors, which can help point to remediation steps.
+
+If your instance often reports unhealthy components but still serves traffic normally, consider using CLI-based probe checks in Kubernetes and apply their [filter feature](#filters).
+
+## CLI
+
+Pomerium images include the pomerium binary, which provides a health subcommand. This command exits with status 1 if the instance is unhealthy, and 0 if healthy.
+
+The CLI health check can be used in environments like Docker Compose and ECS. See the [Compose documentation](https://docs.docker.com/reference/compose-file/services/#healthcheck) and [ECS documentation](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_HealthCheck.html) for details.
+
+```
+Usage:
+  pomerium health [flags]
+
+Flags:
+  -e, --exclude stringArray   list of health checks to exclude from consideration
+  -a, --health-addr string    port of the pomerium health check service (default "127.0.0.1:28080")
+  -h, --help                  help for health
+  -v, --verbose               prints extra health information
+```
+
+An example configuration for Pomerium in docker compose looks like:
+
+```yaml
+healthcheck:
+  interval: 30s
+  retries: 5
+  test:
+    - CMD
+    - pomerium
+    - health
+  timeout: 5s
+```
+
+In Kubernetes you can deploy like:
+
+```yaml
+livenessProbe:
+  containers:
+    - name: pomerium
+      # ...
+      livenessProbe:
+      exec:
+        command:
+          - pomerium
+          - health
+          - a
+          - $(POD_IP):28080
+      initialDelaySeconds: 5
+      periodSeconds: 30
+      timeoutSeconds: 5
+```
+
+### Filters
+
+Unlike HTTP probes, the CLI provides additional flexibility through exclude filters, which let you ignore specific internal conditions reported by Pomerium when evaluating health.
```

Comment:
How about adding a check definition section with short descriptions of what each of these checks covers, so they can figure out what they might want to filter out?

Reply:
Yeah probably a good idea to do that, although they're internal checks so may not make intuitive sense to users, so let me take some time to think about the best way to document those